prov/rxm/src/rxm_ep.c

```diff
@@ -1530,6 +1531,18 @@ rxm_ep_conn_progress_deferred_queue(struct rxm_ep *rxm_ep,
 		case RXM_DEFERRED_TX_SAR_SEG:
 			ret = rxm_ep_progress_sar_deferred_segments(def_tx_entry);
 			break;
+		case RXM_DEFERRED_TX_ATOMIC_RESP:
+			ret = fi_send(def_tx_entry->rxm_conn->msg_ep,
```

Comment:
Is this going to generate a completion that the user isn't expecting since this is an emulated software atomic that the user shouldn't be aware of.

Reply:
Like the other case, it can generate a core provider completion but the RxM completion handler will use protocol state to know it should consume the completion entry, release resources, and not write an associated RxM TX CQ entry.

Reply:
I've modified the code to use fi_sendmsg() here so that I can force a completion if the endpoint has been configured for FI_SELECTIVE_COMPLETION. Note that the completion will not reported to the application, but is used by the protocol to cleanup the buffer resources.

Reply:
Ok.